DEV_GUIDE.md

```diff
@@ -19,7 +19,10 @@ mvn xml-format:xml-format fmt:format
 
 ## Java Version 
 
-Stargate currently runs on Java 8 due to its backend dependencies. It's important to ensure that you have the correct JDK 8 installed before you can successfully compile the Stargate project. There are a number of versions of JDK 8 and a number of different ways to install them, but not all of them will work successfully with Stargate.
+Stargate uses multiple JDKs for its various components, as described in the sections below. 
+
+### Coordinator node
+The coordinator currently runs on Java 8 due to its backend dependencies. It's important to ensure that you have the correct JDK 8 installed before you can successfully compile the Stargate project. There are a number of versions of JDK 8 and a number of different ways to install them, but not all of them will work successfully with Stargate.
```

Comment:
If we state this (which makes sense) we also need to indicate version(s) -- likely a known minimum version? -- that works. Or, if we know of problems with older versions, maybe list ones we know of.

In a way I wonder if we should at first say "Java 8, suggested minimum version `1.8.0_332`" unless and until we hear of actual problems (I agree it is possible some version won't work but not sure how common such problems are).

Reply:
I don't know what the known minimum version would be?

Reply:
I'm adding a reference to the CI workflow where someone could check which JDK we're currently testing against

Reply:
Uhh. We seem to use floating version "8", i.e. latest 8.0.x. Same for 17.

So just need to select some values; locally I use -- for example 

* 8.0.345 (earlier .292 and .332 which seemed to work)
* 17.0.4 (17.0.2 seemed fine earlier too).